bindings/python/src/lib.rs

```diff
@@ -0,0 +1,318 @@
+#![feature(specialization, const_fn)]
+extern crate jail;
+extern crate pyo3;
+extern crate rctl;
+
+use std::collections::HashMap;
+
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyInt, PyString};
+use pyo3::PyObjectWithToken;
+use pyo3::{exceptions, PyDowncastError};
+
+use jail as native;
+
+#[pyclass]
+struct JailError {
+    inner: native::JailError,
+    token: PyToken,
+}
+
+#[pymethods]
+impl JailError {
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("{}", self.inner))
+    }
+}
+
+#[pyclass]
+struct RunningJail {
+    inner: native::RunningJail,
+    // As Python keeps its own reference counter, and we can't be sure that
+    // that reference will be destroyed when we stop or kill the Jail, we have
+    // to keep track of that ourselves.
+    dead: bool,
+    token: PyToken,
+}
+
+#[pymethods]
+impl RunningJail {
+    #[new]
+    fn __new__(obj: &PyRawObject, jid: i32) -> PyResult<()> {
+        obj.init(|token| RunningJail {
+            inner: native::RunningJail::from_jid(jid),
+            dead: false,
+            token,
+        })
+    }
+
+    /// Return a String representation of the Jail
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("{:?}", self.inner))
+    }
+
+    #[getter]
+    /// The Jail ID
+    fn get_jid(&self) -> PyResult<i32> {
+        Ok(self.inner.jid)
+    }
+
+    #[getter]
+    fn get_ips(&self) -> PyResult<Vec<String>> {
+        Ok(self
+            .inner
+            .ips()
+            .map_err(|_| exceptions::SystemError::py_err("Could not get IP Addresses"))?
+            .iter()
+            .map(|addr| format!("{}", addr))
```

Comment:
might be a bit too much to ask for, but you could return actual IP addresses here

Reply:
In what type?

Reply:
https://docs.python.org/3/library/ipaddress.html

Reply:
I'd like to keep support for python 2 - at least for now. But I guess I'd have to construct that from a string anyways?